packages/react-diagrams-core/src/states/DragNewLinkState.ts

```diff
@@ -92,10 +92,13 @@ export class DragNewLinkState extends AbstractDisplacementState<DiagramEngine> {
 	 */
 	fireMouseMoved(event: AbstractDisplacementStateEvent): any {
 		const portPos = this.port.getPosition();
-		const engineOffsetX = this.engine.getModel().getOffsetX();
-		const engineOffsetY = this.engine.getModel().getOffsetY();
-		const linkNextX = portPos.x - engineOffsetX + (this.initialXRelative - portPos.x) + event.virtualDisplacementX;
-		const linkNextY = portPos.y - engineOffsetY + (this.initialYRelative - portPos.y) + event.virtualDisplacementY;
+		const zoomLevelPercentage = this.engine.getModel().getZoomLevel() / 100;
+		const engineOffsetX = this.engine.getModel().getOffsetX() / zoomLevelPercentage;
+		const engineOffsetY = this.engine.getModel().getOffsetY() / zoomLevelPercentage;
+		const initialXRelative = this.initialXRelative / zoomLevelPercentage;
+		const initialYRelative = this.initialYRelative / zoomLevelPercentage;
+		const linkNextX = portPos.x - engineOffsetX + (initialXRelative - portPos.x) + event.virtualDisplacementX;
+		const linkNextY = portPos.y - engineOffsetY + (initialYRelative - portPos.y) + event.virtualDisplacementY;
```

Comment:
will accept this for now, but I think most of this data is available already in the displacement state in some form or another.

Reply:
the only things that take into consideration the zoomLevel in the AbstractDisplacementState class are virtualDisplacementY and virtualDisplacementY. Maybe the zoomLevel can be considered in the AbstractDisplacementState's  initialX/Yrelaive instead of in the DragNewLinkState